BLSQ/iaso
# IA-2697: Change request UI for org unit creation

Need specific UI for change request that create a new org unit

Related JIRA tickets : IA-2697

## Self proofreading checklist


## Doc
Code seems self-explanatory enough

## Changes

- use `getRowProps` to set the row background color to yellow if no org_unit_id in main list view
- use a default message for the org unit name cell when no `org_unit_name` in main list view
- Add `isNewOrgUnit` prop in `ReviewOrgUnitChangeButtons`, `ReviewOrgUnitChangeDetailsHead` and `OrgUnitChangeDetaislRow`
- Compute `isNewOrgUnit` value in `ReviewOrgUnitChangesDialog`
- Add check on `name` and `org_unit_type`_fields in Dialog
- Add placeholder row for when name or type is missing in Dialog table
- Display placeholder row in Dialog table if new org unit and name and/or type is missing
- Disable "approve" button in Dialog table if new org unit and name or type is missing 


## How to test

Because we're handling cases that the backend doesn't allow yet (org unit change request without org unit), the booleans triggering the UI chnages need to be set to `true` in the code for the changes to be visible:

- In `ReviewOrgUnitChangesDialog`, set `isNewOrgUnit` to `true`
- Open a `new` change request:
    - There should be only one column for changes
    - The text of the dialog title and confirm button should change
- Now set `hasName` and/or `hasType` to `false`:
    - There should be a placeholder row for name and/or org unit type, with no checkbox, and an error message
    -  The save button should always stay disabled
 - Go to `ReviewOrgUnitChangesTable`. In the `getRowProps` method, remove the `if`statement:
    - All rows should have a yellow backgroundColor in main list table
- Go to `useColumns`: comment out the `LinkToOrgUnit`
    - There should be a placeholder message instead of the link to the org unit 

The check to determine if we're creating an org unit is on the `org_unit` field of the change request.

## Print screen / video
![Screenshot 2024-02-13 at 12 19 05](https://github.com/BLSQ/iaso/assets/38907762/0b7c40bf-2095-4502-832c-135489b67e20)
![Screenshot 2024-02-13 at 12 20 06](https://github.com/BLSQ/iaso/assets/38907762/d651e425-a1e6-43ac-b2ca-23ae3cbb4d39)
